[Feature-10044][dolphinscheduler-task-http] Implements HTTP parameter passing to the next node

ITBOX-ITBOY opened the pull request:
The submitted code implements HTTP parameter passing to the next node.

close #10044

devosend committed:
fix spcial week init bug (#10183)

Commit:
Feature-10044][dolphinscheduler-task-http] Implement HTTP Body to pass parameters down

SbloodyS commented:
Hi @ITBOX-ITBOY , to target specific issue from PR, we recommend use keyword fix: #issue_id or close: #issue_id or closes: #issue_id in your PR describe(not title, jus desc). It would not only connect issue to PR but also close issue automatically when PR is be closed.

BTW, please also submit the modification of the document in this PR.

longtb976 committed:
[Fix-10076] queryUserByToken use server timezone (#10177)

* [Fix-10076] queryUserByToken use server timezone

* [Fix-10076] queryUserByToken use server timezone

Amy0104 commented on dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-http.ts:
```diff
@@ -192,7 +192,7 @@ export function useHttp(model: { [field: string]: any }): IJsonItem[] {
       model,
       field: 'localParams',
-      isSimple: true
+      isSimple: false
```
Are you sure you need the direct and the type here?

labbomb replied:
+1

ITBOX-ITBOY replied:
Yes, because the front end displays drop-down selections for in/out and VARCHAR

QuakeWang committed:
[doc] Refactor global parameter docment (#10045)

EricGao888 committed:
[doc] Update project directory structure in docs (#10107)

SbloodyS committed:
[Doc] Add complie supporting system (#10188)

songjianet committed:
[Fix][UI] Fix data-quality component multi-language support. (#10202)

* [Fix][UI] Fix data-quality component multi-language support.

* [Fix][UI] Fix data-quality component multi-language support.

SnowMoon-Dev committed:
[doc] Update delete workflow description (#10203)

SnowMoon-Dev committed:
[Fix-10199] [Workflow/Workflow Definition] After deleting messages in batches, the error message is repeated (#10201)

* remove duplicate prompts

* update import package

SbloodyS commented on dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/main/java/org/apache/dolphinscheduler/plugin/task/http/HttpParameters.java:
```diff
@@ -130,4 +136,44 @@ public int getSocketTimeout() {
+            return;
+        }
+        Map<String, String> httpMapByString = getHttpMapByString(result);
+        //判断是否为空
```
Please use english instead of chinese

ITBOX-ITBOY replied:
Already processed, code already committed

SbloodyS commented on dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/main/java/org/apache/dolphinscheduler/plugin/task/http/HttpParameters.java:
```diff
@@ -130,4 +136,44 @@ public int getSocketTimeout() {
+                updateRM.put(info.getProp(), updateResult);
+                updateRL.add(updateRM);
+                result = JSONUtils.toJsonString(updateRL);
+                // break;
```
We should remove this.

ITBOX-ITBOY replied:
This part of code cannot be deleted because updateResult gets the body value, info.getProp() gets the output key from localParm, {"body":{"userName":"testName"}}

SbloodyS replied:
What i mean is remove ```// break;```

ITBOX-ITBOY replied:
Break has been removed and the code has been committed

SbloodyS commented on dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/parameters/AbstractParameters.java:
```diff
@@ -201,4 +202,27 @@ private void addPropertyToValPool(Property property) {
+     * @param result
+     * @return
+     */
+    public static Map<String, String> getHttpMapByString(String result) {
```
I think this method ```getHttpMapByString``` should be included in ```HttpTask``` instead of Abstract class.

ITBOX-ITBOY replied:
GetHttpMapByString is moved to HttpParameters because it's used in this

jieguangzhou committed:
[Feature][MLops]  Add MLflow Projects (custom project) component (#10175)

* Add MLflow Projects (custom project) component

* fix something

* update document

* update document

* Supplementary test cases

EricGao888 committed:
[Bug][Deploy]Fix worker-server path in script scp-hosts.sh (#10208) (#10209)

SnowMoon-Dev committed:
fix#10214 Delete online workflows in batches, prompt information is accurate (#10215)

zhongjiajie committed:
[python] Fix process definition attr release state not work (#10151)

This patch fix the error release state not work when it set to
offline and submit it to Java gateway, it error because we do
not pass the attribute to Java gateway function
`createOrUpdateProcessDefinition`

close: #9779

rockfang committed:
[Fix][UI] add assignment of totalPage (#10165)

zhongjiajie committed:
[doc] Update task DataX document (#10218)

qingwli committed:
[Fix-10103][k8s]Fix k8s Change DataSource Error (#10128)

songjianet committed:
[Fix][UI] Fix the problem of multi-language support of workflow instance host in Chinese state. (#10223)

Commit:
1.Modify Chinese comments to English
2.Move AbstractParameters' getHttpMapByString method to HttpParameters

Commit:
1.Delete the break

SbloodyS commented on dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/parameters/AbstractParameters.java:
```diff
@@ -17,6 +17,7 @@
 
 package org.apache.dolphinscheduler.plugin.task.api.parameters;
 
+import com.fasterxml.jackson.core.type.TypeReference;
```
Please remove unnessnary change.

ITBOX-ITBOY replied:
Already processed, code already committed

SbloodyS commented on dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/parameters/AbstractParameters.java:
```diff
@@ -201,4 +202,5 @@ private void addPropertyToValPool(Property property) {
         varPool.removeIf(p -> p.getProp().equals(property.getProp()));
         varPool.add(property);
     }
+
```
same here.

ITBOX-ITBOY replied:
Already processed, code already committed

Commit:
1.Delete invalid Spaces and imported packages

qingwli committed:
[doc] Update kubernetes' FAQ (#10221)

SbloodyS commented on dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/parameters/AbstractParameters.java:
```diff
@@ -17,23 +17,17 @@
-
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.node.ArrayNode;
+import java.util.*;
```
Please avoid import ```*```

ITBOX-ITBOY replied:
Already processed, code already committed

QuakeWang committed:
[doc] Refactor local parameter docment (#10119)

Commit:
1.Handle the problem of importing package as *

SbloodyS commented on dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/parameters/AbstractParameters.java:
```diff
@@ -17,23 +17,22 @@
 
 package org.apache.dolphinscheduler.plugin.task.api.parameters;
 
+import com.fasterxml.jackson.databind.JsonNode;
```
Please remove unnessnary change in this file.

ITBOX-ITBOY replied:
Already processed, code already committed

Commit:
1.Handle unnecessary changes

SbloodyS commented:
> Hi @ITBOX-ITBOY , to target specific issue from PR, we recommend use keyword fix: #issue_id or close: #issue_id or closes: #issue_id in your PR describe(not title, jus desc). It would not only connect issue to PR but also close issue automatically when PR is be closed.
> 
> BTW, please also submit the modification of the document in this PR.

Please add the docs to this PR. Thanks. @ITBOX-ITBOY

SnowMoon-Dev committed:
[fix-10207] When the registered alarm plugin is empty, the error message is not clear (#10224)

* Improve the problem that the error message is not clear when the registered alarm plugin is empty

* update error info

* update error info

* update error info

* Update Status.java

fix typo

Co-authored-by: xiangzihao <460888207@qq.com>

ITBOX-ITBOY commented:
> > Hi @ITBOX-ITBOY , to target specific issue from PR, we recommend use keyword fix: #issue_id or close: #issue_id or closes: #issue_id in your PR describe(not title, jus desc). It would not only connect issue to PR but also close issue automatically when PR is be closed.
> > BTW, please also submit the modification of the document in this PR.
> 
> Please add the docs to this PR. Thanks. @ITBOX-ITBOY

Document links
https://github.com/apache/dolphinscheduler/issues/10044

SbloodyS commented:
> > > Hi @ITBOX-ITBOY , to target specific issue from PR, we recommend use keyword fix: #issue_id or close: #issue_id or closes: #issue_id in your PR describe(not title, jus desc). It would not only connect issue to PR but also close issue automatically when PR is be closed.
> > > BTW, please also submit the modification of the document in this PR.
> > 
> > 
> > Please add the docs to this PR. Thanks. @ITBOX-ITBOY
> 
> Document links #10044

We should add this to the official document instead of putting in issue.
```docs/docs/zh/guide/task/http.md```
```docs/docs/en/guide/task/http.md```

songjianet committed:
[Fix][UI] Fix issue with treemap depth in workflow relationships. (#10229)

* [Fix][UI] Fix issue with treemap depth in workflow relationships.

* [Fix][UI] Fix issue with treemap depth in workflow relationships.

Commit:
1.Documentation to improve the HTTP pass-down function

ITBOX-ITBOY commented:
> > > > Hi @ITBOX-ITBOY , to target specific issue from PR, we recommend use keyword fix: #issue_id or close: #issue_id or closes: #issue_id in your PR describe(not title, jus desc). It would not only connect issue to PR but also close issue automatically when PR is be closed.
> > > > BTW, please also submit the modification of the document in this PR.
> > > 
> > > 
> > > Please add the docs to this PR. Thanks. @ITBOX-ITBOY
> > 
> > 
> > Document links #10044
> 
> We should add this to the official document instead of putting in issue. `docs/docs/zh/guide/task/http.md` `docs/docs/en/guide/task/http.md`

The functional documentation has been written and submitted
\docs\docs\en\guide\parameter\context.md
\docs\docs\zh\guide\parameter\context.md

qingwli committed:
[doc] Delete SPRING_DATASOURCE_DRIVER_CLASS_NAME in doc (#10228)

qingwli committed:
[improve] Change Mysql Driver (#10220)

zhongjiajie committed:
[fix] Task group queue query all list error (#10237)

Add support query all task queue by not given specific groupId

close: #10102

zhuangchong reviewed (commented):
Tasks like shell and sql both support input and output parameters. I hope they can be put into a tool class for processing. I hope you can make some optimizations in the next PR. Finally, it is recommended to use the latest version of the UI for the pictures on the document.

SnowMoon-Dev committed:
[Fix#9923] The alarm instance management interface is always in the loading state (#9924)

* fix alert plugin loading bug

* update use-form.ts

* remove unused package

* Update use-form.ts

* Update use-form.ts

devosend committed:
[Fix][UI] Set task group name empty of default value (#10243)

zhongjiajie committed:
[fix][deploy] File overwrite when specific value set to installPath (#10241)

When our `installPath` in file `install_env.sh` set to `/`
or empty or related path to `/` some system specific directory
, like `/bin` and `libs`, will be overwritten, and it is not expected

zhongjiajie committed:
[doc] Add setValue syntax in local param (#10245)

ref: #9576
same as： #9671

zhongjiajie committed:
[fix] Deploy scp-hosts use fullpath avoid dangling (#10249)

This path fix error modify worker application.yaml when
running `scp-hosts.sh` script in different directory. Currently
setting only work when running `scp-hosts.sh` in path
`<DOLPHINSCHEDULER_HOME>/bin` using command `./scp-hosts.sh`.
And will fail when we run script in directory
`<DOLPHINSCHEDULER_HOME>` using command `./bin/scp-hosts.sh`

ref: #10209 and #10208

caishunfeng committed:
[Bug] init docker compose DATABASE env (#10244)

skip source env if in docker

zhongjiajie committed:
[doc] Correct all setup documents (#10074)

* Correct the login page URL
* change version and database configuration
* change setup and upgrade

Commit:
[Fix-10181] Fix the logic of judging that the tenant does not exist (#10185)

* [Fix-10181] Fix the logic of judging that the tenant does not exist

Use the linux command as id to get the user information that exists in /etc/passwd file and the cached sssd user.
for example:
id test
1. exist in /etc/passwd file or ldap :  uid=1030(test) gid=1030(test) groups=1030(test)
2. no exist  in /etc/passwd file and ldap: id: test: no such user

Temporarily unable to test the system for windows and mac

* [Fix-10181] Fix the logic of judging that the tenant does not exist

Use the linux command as id to get the user information that exists in /etc/passwd file and the cached sssd user.
for example:
id test
1. exist in /etc/passwd file or ldap :  uid=1030(test) gid=1030(test) groups=1030(test)
2. no exist  in /etc/passwd file and ldap: id: test: no such user

Temporarily unable to test the system for windows and mac

* [Fix-10181] Fix the logic of judging that the tenant does not exist

Use the linux command as id to get the user information that exists in /etc/passwd file and the cached sssd user.
for example:
id test
1. exist in /etc/passwd file or ldap :  uid=1030(test) gid=1030(test) groups=1030(test)
2. no exist  in /etc/passwd file and ldap: id: test: no such user

Temporarily unable to test the system for windows and mac

* [Fix-10181] Fix the logic of judging that the tenant does not exist

The configuration item adds 'tenant-distributed-user' in worker application.yaml to make it suitable for distributed users. If it is false, the original logic remains unchanged.

At present, considering that it is a distributed user, it should not be allowed to create users in linux

Use the linux command as id to get the user information that exists in /etc/passwd file and the cached sssd user.
for example:
id test
1. exist in /etc/passwd file or ldap :  uid=1030(test) gid=1030(test) groups=1030(test)
2. no exist  in /etc/passwd file and ldap: id: test: no such user

Temporarily unable to test the system for windows and mac

* [Fix-10181] Fix the logic of judging that the tenant does not exist

Add test method

The configuration item adds 'tenant-distributed-user' in worker application.yaml to make it suitable for distributed users. If it is false, the original logic remains unchanged.

At present, considering that it is a distributed user, it should not be allowed to create users in linux

Use the linux command as id to get the user information that exists in /etc/passwd file and the cached sssd user.
for example:
id test
1. exist in /etc/passwd file or ldap :  uid=1030(test) gid=1030(test) groups=1030(test)
2. no exist  in /etc/passwd file and ldap: id: test: no such user

Temporarily unable to test the system for windows and mac

* [Fix-10181] Fix the logic of judging that the tenant does not exist

Add parameter description to configuration.md

Add test method

The configuration item adds 'tenant-distributed-user' in worker application.yaml to make it suitable for distributed users. If it is false, the original logic remains unchanged.

At present, considering that it is a distributed user, it should not be allowed to create users in linux

Use the linux command as id to get the user information that exists in /etc/passwd file and the cached sssd user.
for example:
id test
1. exist in /etc/passwd file or ldap :  uid=1030(test) gid=1030(test) groups=1030(test)
2. no exist  in /etc/passwd file and ldap: id: test: no such user

Temporarily unable to test the system for windows and mac

* [Fix-10181] Fix the logic of judging that the tenant does not exist

Add parameter description to configuration.md

Add test method

The configuration item adds 'tenant-distributed-user' in worker application.yaml to make it suitable for distributed users. If it is false, the original logic remains unchanged.

At present, considering that it is a distributed user, it should not be allowed to create users in linux

Use the linux command as id to get the user information that exists in /etc/passwd file and the cached sssd user.
for example:
id test
1. exist in /etc/passwd file or ldap :  uid=1030(test) gid=1030(test) groups=1030(test)
2. no exist  in /etc/passwd file and ldap: id: test: no such user

Temporarily unable to test the system for windows and mac

* [Fix-10181] Fix the logic of judging that the tenant does not exist

Add parameter description to configuration.md

Add test method

The configuration item adds 'tenant-distributed-user' in worker application.yaml to make it suitable for distributed users. If it is false, the original logic remains unchanged.

At present, considering that it is a distributed user, it should not be allowed to create users in linux

Use the linux command as id to get the user information that exists in /etc/passwd file and the cached sssd user.
for example:
id test
1. exist in /etc/passwd file or ldap :  uid=1030(test) gid=1030(test) groups=1030(test)
2. no exist  in /etc/passwd file and ldap: id: test: no such user

Temporarily unable to test the system for windows and mac

Co-authored-by: ouyangl <ouyangl@tebon.com.cn>

songjianet committed:
[Style][UI] Split multilingual modules. (#10256)

songjianet committed:
[Fix][UI] Fix status statistics card height is inconsistent. (#10268)

106umao committed:
[Fix] Fix the timeout alarm button is invalid (#10260)

WangJPLeo committed:
[Feature][Permission]  Project management module authority control refactoring, authority abstract interface setting. (#10261)

* Create process definition with task group and task group priority cannot be save into db

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* [Feature][API] Permission control of project

* The result of the assertion cannot be changed. The only difference is the judgment condition.

* test recovery.

* controller test fix.

* ProjectE2ETest fix.

* WebElement import.

* Prioritize problem solving.

* Judging the length of the set fix

Co-authored-by: houshitao <shitaohou@163.com>
Co-authored-by: hstdream <33045461+hstdream@users.noreply.github.com>

106umao committed:
[Fix] Fix the line.separator problem for alert-plugin' test case #10277 (#10278)

* [Fix] Fix the line.separator problem for alert-plugin' test case

* [Fix] Fix the line.separator problem for alert-plugin' test case

cadl committed:
[fix] Fix the  SERIAL_DISCARD execution_type not working (#10258)

* fix always stop process_instance when execution_type is SERIAL_DISCARD

* test: no running_process, submit it

QuakeWang committed:
[doc] Upgrade screenshot to latest version in dev document (#10242)

* fix issue-9898
* delete no use pictures

QuakeWang committed:
[doc] Refactor context parameter docment (#10227)

GavinGYM committed:
[docs] Added local file configuration guide for resource center (#10264)

* Added Local File Resource Configuration Guide to the document.
* Removed contents with windows features in the documents and improved expression.
* Specify `the user who deploy dolphinscheduler have read and write permissions` in en and zh docs.

Co-authored-by: xiangzihao <460888207@qq.com>

Amy0104 committed:
[Fix][UI] Add mapping_columns filed for the data_quality ruleInputParamter. (#10286)

Tianqi-Dotes committed:
[Feature][Datasource]replace DataSourceController API with string JSON and cast to DTO (#10276)

* replace DataSourceController API with string JSON and cast to DTO

* add DataSourceProcessorManager and DataSourceProcessorProvider to datasource processor functions

SnowMoon-Dev committed:
[doc] Update the development environment to build the software version (#10246)

zhongjiajie commented on docs/docs/en/guide/parameter/context.md:
```diff
@@ -64,3 +64,25 @@ For example, in the figure below:
+
+Step 1: Drag an HTTP task, fill IN body for KEY, select OUT for IN/OUT, select VARCHAR for output data type, always select VARCHAR, nothing else.
+
+<img src="/img/httpParam/1-1.png" alt="1-1" style="zoom:50%;" />
```
* use markdown syntax when img using, ex `[httpParam1](/img/httpParam1-1.png)`. same as other imgs
* please use the English version of screenshot
* change the img file to more meaningful name instead of `1-1.png` and `1-2.png`, maybe you could directly using `/img/httpparam-1.png` and so on

ITBOX-ITBOY replied:
It has been resolved and the code has been committed

zhongjiajie reviewed (commented):
and could you add some unit tests on this feature?

SbloodyS committed:
[doc] Fix api standard error (#10292)

* fix api standard error
* add patch method

songjianet committed:
[Fix][UI] Display the result page when the DB page has no data. (#10298)

zhongjiajie committed:
[chore] update by release version (#10299)

hstdream committed:
[Feature][API] Modify the permissions of project management, security center, data source center and data quality module. (#10289)

* Resource userName is null

* [Fix-10082]Resource userName is null and udf function manager add a username

* [Fix-10082]Resource userName is null and udf function manager add a username

* [Fix-10082]Resource userName is null and udf function manager add a username

* [Feature][API] Modify the permissions of project management, security center, data source center and data quality module.

* [Feature][API] Modify the permissions of project management, security center, data source center and data quality module.

* [Feature][API] Modify the permissions of project management, security center, data source center and data quality module.

Co-authored-by: houshitao <shitaohou@163.com>

WangJPLeo committed:
[Feature][Permission] Reconstruction of permissions of resource center and monitoring center. (#10307)

* Reconstruction of permissions of resource center and monitoring center.

* clear local logs.

* resource query fix

vagetablechicken committed:
[feat][task] Add OpenMLDB task plugin (#10198)

EricGao888 committed:
[Feature][Task Plugin] Add support for dynamic form for zeppelin task plugin (#9977) (#10269)

brave-lee committed:
[Fix-8828] [Master] Assign tasks to worker optimization (#9919)

* fix 9584

* master recall

* fix ut

* update logger

* update delay queue

* fix ut

* remove sleep

Co-authored-by: 进勇 <lijinyong@cai-inc.com>
Co-authored-by: JinyLeeChina <jiny.li@foxmail.com>

ruanwenjun committed:
Fix database initialize late in standalone. (#10296)

SbloodyS committed:
[Feature-10290] [API] Refactor org.apache.dolphinscheduler.api.controller.AccessTokenController#createToken api (#10301)

* add feature_10290

* fix dolphinscheduler-ui/pnpm-lock.yaml error

* fix dolphinscheduler-ui/pnpm-lock.yaml error

* fix incorrect dependency

* fix incorrect dependency and unit test

* fix unit test error

* add dolphinscheduler-api/logs to .gitignore

* add AccessTokenV2ControllerTest.java

* fix AccessTokenV2ControllerTest error

kezhenxu94 committed:
Clean up unused dependencies and packaging issues (#9944)

songjianet committed:
[Style][UI] Format code. (#10314)

zhongjiajie committed:
[ci] Fix mergeable for ddl change and remove label unexpect (#7099)

aiwenmo committed:
[Fix-10317] [ui] Fix the tips of startTimeout in the jupyter form (#10319)

Co-authored-by: wenmo <32723967+wenmo@users.noreply.github.com>

EricGao888 committed:
[Bug][Api-Server]Fix bug when user without tenant uploads file in resource center (#10303)

QuakeWang committed:
[doc] Add how to use resource in shell task (#10308)

WangJPLeo committed:
[Fix] Privilege Reconstruction Data Query Fix & Resource Creation Post Oper… (#10313)

* Privilege Reconstruction Data Query Fix & Resource Creation Post Operations.

* Resource Retry

* e2e retry

* create project log add

* project list query log add

* clean project log

* delete delay

* delete delay

* remove post handle

* project e2e

* browser refresh

* browser refresh

* e2e fix

* e2e browser refresh

* rowk flow e2e fix

* mapper deduplication

* udf e2e

* e2e

songjianet committed:
[Fix][UI] Fixed the issue that the status statistics on the project overview page were inconsistent. (#10328)

WangJPLeo committed:
[Feature] Should not add refresh logic in the e2e case. (#10331)

* Should not add refresh logic in the e2e case.

* When the returned data from paging is empty, it should be an empty collection.

* resource query fix.

* e2e rerun

* timeZone ut fix

* udf resource check fix

* code deduplication

* rerun

guodongym committed:
[Fix-10305][common] AWS config Constant definitions are repeated (#10311)

SbloodyS committed:
fix bug_10343 (#10344)

SbloodyS committed:
[CI] bumping actions/cache to v3 to fix ci error (#10345)

* bumping actions/cache to v3

* fix JSONUtilsTest.java unit test error

songjianet committed:
[Build][UI] Upgrade dependencies. (#10346)

* [Build][UI] Upgrade dependencies.

* [Build][UI] Upgrade dependencies.

* [Build][UI] Upgrade dependencies.

SbloodyS committed:
[CI] Add code owner of .github (#10347)

* add code owner

* add code owner

* add code owner

guodongym committed:
[Fix-10318][ui] Fix EMR Task UI custom parameters Is not correct (#10334)

* Modify useCustomParams isSimple: true

ruanwenjun committed:
Add dolphinscheduler-task-all in api module (#10350)

zhongjiajie committed:
[doc] Use related path for img (#10325)

Use related path in our docs for imgs, previous
can not use because website need the absolute path
from root directory, and after we merged
apache/dolphinscheduler-website#789 we have covert
function to do that

close: #9426

ruanwenjun committed:
Move datasource params to datasouce plugin module (#10353)

* Move datasource params to datasouce plugin module

pjfanning committed:
issues-10354: upgrade commons-io to fix CVE (#10355)

pjfanning committed:
issue-10356: upgrade logback to fix cve (#10357)

ruanwenjun committed:
Remove quartz at WorkerServer (#10358)

* Remove quartz at WorkerServer

* move k8s and permission from dolphinscheduler-service to dolphinscheduler-api

ruanwenjun committed:
Add dolphinscheduler-scheduler module (#10360)

* Add dolphinscheduler-scheduler module

CalvinKirs committed:
[Docs]Update Slack Invite Link (#10362)

WangJPLeo committed:
[Fix-10338] Tenant does not belong to resource permission. (#10339)

* Tenant does not belong to resource permission.

* rerun

* rerun again

* resource create and upload fix.

* func constant clear.

zhaohehuhu committed:
[improve] Using create or replace function in sql task (#10170)

ruanwenjun committed:
[Fix] [Worker] Fix worker will hang if fails to start (#10342)

* Fix worker will hang if fails to start

* Add .run to ignore

retime123 committed:
[Feature][Monitoring] Added disk available capacity monitoring (#10287)

* Added disk available capacity monitoring

* resolve conflict
Added disk available capacity monitoring

* resolve conflict
Added disk available capacity monitoring

* resolve conflict
Added disk available capacity monitoring

songjianet committed:
[Docs] Upgrade license. (#10365)

huangchenguang123 committed:
[doc] Add switch task demo (#10247)

Co-authored-by: QuakeWang <45645138+QuakeWang@users.noreply.github.com>

SbloodyS committed:
[Feature-6758][Task] Add limit resource usage for tasks base on cgroup (#10373)

ruanwenjun committed:
Optimize MasterServer, add MasterRPCService (#10371)

* Optimize MasterServer, avoid NPE

ruanwenjun committed:
[Feature][metrics] Add master, worker metrics (#10326)

* Add mater metrics

* fix UT

* Add url to mysql profile

* Add worker metrics

* Update grafana config

* Add system metrics doc

* Add process failover counter

* Add metrics image

* Change jpg to png

* Add command insert metrics

* Fix UT

* Revert UT

SbloodyS committed:
change to common lang3 and fix checkstyle (#10380)

Hwting committed:
[fix][deploy]fix run on kubernetes bug (#10379)

* [fix][kubernetes]fix run on kubernetes bug

Fix the problem that the "REGISTRY_ZOOKEEPER_CONNECT_STRING" variable zookeeper port number is "nil"

* [fix][kubernetes]fix run on kubernetes bug

Fix the problem that the "REGISTRY_ZOOKEEPER_CONNECT_STRING" variable zookeeper port number is "nil"

pjfanning committed:
Upgrade log4j in E2E (#10359)

SbloodyS committed:
[CI] Add some codeowner (#10385)

* add some codeowner

* add some codeowner

WangJPLeo committed:
[Fix-10367] Permission check related key adjustment. (#10368)

* Permission check related key adjustment.

* resource create or upload fix.

* Analysis Permission fix.

* Ut coverage add.

* rerun

* source permission fix.

* ut coverage is achieved

* Reference permission package path modification

* k8s path bug fix.

* k8s path bug fix.

* Corrected controller method name and rerun.

jieguangzhou committed:
[Feature] [MLOps] support mlflow deploy with docker compose (#10217)

* [Feature] [MLOps] support mlflow deploy with docker compose

fix doc

Update docs/docs/en/guide/task/mlflow.md

fix doc

Co-authored-by: Jiajie Zhong <zhongjiajie955@gmail.com>

revert cancel modification

fix ENV name and docker compose command

* fix doc image link

* fix testModelsDeployDockerCompose

* add docker compose container health check and fix mlflow bug

* update docker compose healthcheck timeout

EricGao888 committed:
[Feature][Metrics] Enable prometheus to collect metrics in standalone mode demo  (#10398)

ruanwenjun committed:
Add some warning log in master (#10383)

* Add some warn log in master

* fix may skip sleep

Commit:
1.Improve the HTTP parameter transfer function documentation
2.Optimized the http-param parameter passing function

Commit:
1.Adding unit tests

Commit:
1.Modify Chinese notes

zhongjiajie committed:
[chore] Remove unnecessary badge total lines (#10405)

total lines badge meaningless to our project and is will cause
doc check failed sometiem

ref: https://github.com/apache/dolphinscheduler/actions/runs/2474962451/attempts/1

Commit:
1.Modify inaccurate text descriptions

hstdream committed:
[Improve] Enhance complement function transformation (#10376)

ITBOX-ITBOY commented:
> 
It has been resolved and the code has been committed

SbloodyS committed:
[Fix-10412] [Standalone] Failed to start standalone server according to the official document (#10414)

* fix bug_10412

* fix bug_10412

WangJPLeo committed:
Fixed the problem of not having permission to modify the allocated resource data. (#10410)

Amy0104 committed:
[Feature][UI] Add manual input for scheduling dates. (#10396)

* [Feature][UI] Add manual input for scheduling dates.

* [Fix][UI] Fix singlequote.

* [Feature][UI] Remove check for duplicate scheduling dates.

SnowMoon-Dev committed:
Update chart.yaml dependencies version (#10420)

ruanwenjun committed:
Add mysql registry plugin (#10406)

* Add mysql registry plugin

CalvinKirs committed:
[CI] Add OWASP Dependency Check (#10058)

hiSandog committed:
[fix][Python] Support same task name in project defferent process definition (#10428)

close: #10431

zhongjiajie committed:
Merge branch 'dev' into dev-http-param

SbloodyS commented:
Please noticed there are many CI failed. @ITBOX-ITBOY

Commit:
1.Modify the incorrect file name
2.Fixed incorrect parameter values passed down after retry after task failure

Commit:
Merge branch 'dev-http-param' of github.com:ITBOX-ITBOY/dolphinscheduler into dev-http-param

zhongjiajie commented:
CI still failed, and please use English version screenshot in our document, you can switch dolphinscheduler lang to English in right top buttom

Commit:
1.Modify the picture to English version

Commit:
1.Modify the picture to English version

Commit:
1.Fix the problem that the image name is incorrect
2.Delete invalid import packages

Commit:
1.Fix the problem that the image name is incorrect
2.Delete invalid import packages

Commit:
1.Delete abnormal document contents

Commit:
1.Delete abnormal document contents

SbloodyS commented:
@zhuangchong @zhongjiajie PTAL. Thanks.

zhongjiajie commented on docs/docs/en/guide/parameter/context.md:
```diff
@@ -76,3 +76,21 @@ The result of Node_mysql is as follows:
+
+Step 1: Drag an HTTP task, fill IN body for KEY, select OUT for IN/OUT, select VARCHAR for output data type, always select VARCHAR, nothing else.
+
+<img src="../../../../img/httpParam/httpParam-1.png" alt="httpParam-1.png" style="zoom:50%;" />
```
```suggestion
![httpParam-1](../../../../img/httpParam/httpParam-1.png)
```

zhongjiajie commented on docs/docs/en/guide/parameter/context.md:
```diff
@@ -76,3 +76,21 @@ The result of Node_mysql is as follows:
+
+Step 2: After adding an HTTP task type node, accept the parameters passed upstream. This time just add it in the Request Parameters section,
+
+<img src="../../../../img/httpParam/httpParam-2.png" alt="httpParam-2.png" style="zoom:50%;" />
```
```suggestion
![httpParam-2](../../../../img/httpParam/httpParam-2.png)
```

zhongjiajie commented on docs/docs/en/guide/parameter/context.md:
```diff
@@ -76,3 +76,21 @@ The result of Node_mysql is as follows:
+It can be anything, it can be body, it can be anything, parameter,value, it must be the key that you pass upstream. In this case, the key that you pass upstream is body, so we use ${body}
+
+The configuration is complete
+<img src="../../../../img/httpParam/httpParam-3.png" alt="httpParam-3.png" style="zoom:50%;" />
```
```suggestion
![httpParam-3](../../../../img/httpParam/httpParam-3.png)
```

zhongjiajie commented on docs/docs/en/guide/parameter/context.md:
```diff
@@ -76,3 +76,21 @@ The result of Node_mysql is as follows:
+<img src="../../../../img/httpParam/httpParam-3.png" alt="httpParam-3.png" style="zoom:50%;" />
+
+Step 3: You can write a test interface to test whether our parameters are passed successfully.
+<img src="../../../../img/httpParam/httpParam-4.png" alt="httpParam-4.png" style="zoom:50%;" />
```
```suggestion
![httpParam-4](../../../../img/httpParam/httpParam-4.png)
```

zhongjiajie commented on docs/docs/zh/guide/parameter/context.md:
```diff
@@ -76,3 +77,22 @@ Node_mysql 运行结果如下：
+
+第一步:拖一个http类型的任务，在自定义参数中KEY的位置填写body ,在IN/OUT的部分选择OUT,在输出数据类型部分选择VARCHAR，一定要选择VARCHAR，不要选择别的哈。
+
+<img src="../../../../img/httpParam/httpParam-1.png" alt="httpParam-1.png" style="zoom:50%;" />
```
```suggestion
![httpParam-1](../../../../img/httpParam/httpParam-1.png)
```

zhongjiajie commented on docs/docs/zh/guide/parameter/context.md:
```diff
@@ -76,3 +77,22 @@ Node_mysql 运行结果如下：
+
+第二步:在添加一个http任务类型的节点，接收上游传递来的参数。这一次只需要在【请求参数】部分进行添加就可以了，
+
+<img src="../../../../img/httpParam/httpParam-2.png" alt="httpParam-2.png" style="zoom:50%;" />
```
```suggestion
![httpParam-2](../../../../img/httpParam/httpParam-2.png)
```

zhongjiajie commented on docs/docs/zh/guide/parameter/context.md:
```diff
@@ -76,3 +77,22 @@ Node_mysql 运行结果如下：
+参数名可以写任意，可以body也可以是别的，类型选择parameter,value的取值一定是你上一个节点设置输出的key,如:我上一个节点设置的输出key是body，那我下一个节点就要这样取值${body}
+
+配置好后的效果
+<img src="../../../../img/httpParam/httpParam-3.png" alt="httpParam-3.png" style="zoom:50%;" />
```
```suggestion
![httpParam-3](../../../../img/httpParam/httpParam-3.png)
```

zhongjiajie commented on docs/docs/zh/guide/parameter/context.md:
```diff
@@ -76,3 +77,22 @@ Node_mysql 运行结果如下：
+<img src="../../../../img/httpParam/httpParam-3.png" alt="httpParam-3.png" style="zoom:50%;" />
+
+第三步:你可以写一个测试接口,来测试咱们的参数是否传递成功。
+<img src="../../../../img/httpParam/httpParam-4.png" alt="httpParam-4.png" style="zoom:50%;" />
```
```suggestion
![httpParam-4](../../../../img/httpParam/httpParam-4.png)
```

zhongjiajie committed:
Update docs/docs/zh/guide/parameter/context.md

zhongjiajie committed:
Update docs/docs/zh/guide/parameter/context.md

zhongjiajie committed:
Update docs/docs/en/guide/parameter/context.md

zhongjiajie committed:
Update docs/docs/en/guide/parameter/context.md

zhongjiajie committed:
Update docs/docs/zh/guide/parameter/context.md

zhongjiajie committed:
Update docs/docs/en/guide/parameter/context.md

zhongjiajie committed:
Update docs/docs/zh/guide/parameter/context.md

zhongjiajie committed:
Update docs/docs/en/guide/parameter/context.md

zhongjiajie reviewed (dismissed):
Others LGTM

zhongjiajie commented:
start the failed ci

ITBOX-ITBOY commented:
> start the failed ci

What can I do for you @zhongjiajie

SbloodyS commented:
@ITBOX-ITBOY Please merge the latest `dev` to this pr's branch to make CI pass.

ITBOX-ITBOY commented:
> @ITBOX-ITBOY Please merge the latest `dev` to this pr's branch to make CI pass.

I don't have the latest code to submit anymore @SbloodyS

SbloodyS commented:
> > @ITBOX-ITBOY Please merge the latest `dev` to this pr's branch to make CI pass.
> 
> I don't have the latest code to submit anymore @SbloodyS

What i mean is merge the latest `dev` branch to the `dev-http-param` branch.

ITBOX-ITBOY commented:
> > > @ITBOX-ITBOY Please merge the latest  to this pr's branch to make CI pass.`dev`
> > 
> > 
> > I don't have the latest code to submit anymore @SbloodyS
> 
> What i mean is merge the latest branch to the branch.`dev``dev-http-param`

You mean you want me to merge the dev branch code into my dev-http-param branch and push it to pass the check @SbloodyS

SbloodyS commented:
> You mean you want me to merge the dev branch code into my dev-http-param branch and push it to pass the check @SbloodyS

Yes. We've already fixed the CI failed.

ITBOX-ITBOY commented:
> > You mean you want me to merge the dev branch code into my dev-http-param branch and push it to pass the check @SbloodyS
> 
> Yes. We've already fixed the CI failed.

The merge is complete and the commit is complete @SbloodyS

SbloodyS commented:
> The merge is complete and the commit is complete @SbloodyS

There is no new commit in this PR. Please check whether it is successful.

Commit:
1.Add spaces to generate new content, which can be pushed

ITBOX-ITBOY commented:
> > The merge is complete and the commit is complete @SbloodyS
> 
> There is no new commit in this PR. Please check whether it is successful.

The merge is complete and the commit is complete @SbloodyS

SbloodyS commented:
> The merge is complete and the commit is complete @SbloodyS

I found that this branch is 27 commits ahead and 149 commits behind apache:dev. It seems you did not merge the latest `dev` branch to this branch.

Commit:
Merge branch 'dev-http-param' of github.com:ITBOX-ITBOY/dolphinscheduler into dev-http-param

ITBOX-ITBOY commented:
> > The merge is complete and the commit is complete @SbloodyS
> 
> I found that this branch is 27 commits ahead and 149 commits behind apache:dev. It seems you did not merge the latest `dev` branch to this branch.

The merge is complete and the commit is complete @SbloodyS

wangjiucheng123 committed:
Merge pull request #1 from apache/dev

Request update

Commit:
Merge branch 'dev' into dev-http-param

Commit:
1.Invoking System.out.println in source code!

Commit:
1.handle '=' is not preceded with whitespace.

ITBOX-ITBOY commented:
@caishunfeng Please help to deal with the CI inspection, thank you

zhongjiajie commented:
I approval to run the CI

wangjiucheng123 committed:
Precede with whitespace modify '=' is not precede with whitespace

wangjiucheng123 committed:
Merge branch 'dev-http-param' of github.com:ITBOX-ITBOY/dolphinscheduler into dev-http-param

 Conflicts:
	dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpParametersTest.java

ITBOX-ITBOY commented:
> I approval to run the CI

Please help to start the inspection procedure @zhongjiajie

ITBOX-ITBOY commented:
@caishunfeng I'm not sure what I need to change now, so that my functions can be successfully merged into the dev branch

Commit:
1.Solve multiple blank line problems

wangjiucheng123 committed:
Merge branch 'apache:dev' into dev

wangjiucheng123 committed:
Merge branch 'dev' into dev-http-param

wangjiucheng123 committed:
Remove useless code

ITBOX-ITBOY commented:
@SbloodyS Help me run the checker

SbloodyS commented:
> @SbloodyS Help me run the checker

Done.

ITBOX-ITBOY commented:
@caishunfeng @songjianet Help me check the error reported by the program. Do you need me to modify it

SbloodyS commented on dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/main/java/org/apache/dolphinscheduler/plugin/task/http/HttpParameters.java:
```diff
@@ -130,4 +138,69 @@ public int getSocketTimeout() {
+
+        for (Property info : outProperty) {
+            for (int i = 0; i < httpMapByString.size() ; i++) {
+                Map<String, String> stringStringMap=httpMapByString.get(i);
```
```suggestion
                Map<String, String> stringStringMap = httpMapByString.get(i);
```

ITBOX-ITBOY replied:
The code has been processed and submitted

ITBOX-ITBOY replied:
The code has been processed and submitted

EricGao888 commented on docs/docs/zh/guide/parameter/context.md:
```diff
@@ -76,3 +77,22 @@ Node_mysql 运行结果如下：
+
+### HTTP
+
+第一步:拖一个http类型的任务，在自定义参数中KEY的位置填写body ,在IN/OUT的部分选择OUT,在输出数据类型部分选择VARCHAR，一定要选择VARCHAR，不要选择别的哈。
```
Please do not use English comma in Chinese docs and try to avoid informal language like `哈`.

ITBOX-ITBOY replied:
The code has been processed and submitted

ITBOX-ITBOY replied:
The code has been processed and submitted

wangjiucheng123 committed:
Reformatting the code

wangjiucheng123 committed:
Modify the HTTP document

ITBOX-ITBOY commented:
@caishunfeng Please help to run the check program

EricGao888 commented:
> @caishunfeng Please help to run the check program

I've restarted the CI.

ITBOX-ITBOY commented:
> > @caishunfeng Please help to run the check program
> 
> I've restarted the CI.

@EricGao888  There are some problems in the inspection. I have a look. I can't handle them

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpParametersTest.java:
```diff
@@ -75,4 +81,25 @@ public void testCheckValues() {
     }
 
+    @Test
+    public void testHttpParams() {
```
Could you please use `Assert` in the UT to verify the expected behavior?

ITBOX-ITBOY replied:
Have to deal with

EricGao888 commented:
> > > @caishunfeng Please help to run the check program
> > 
> > 
> > I've restarted the CI.
> 
> @EricGao888 There are some problems in the inspection. I have a look. I can't handle them

@ITBOX-ITBOY Did you try to run the UT locally for the http task?

wangjiucheng123 committed:
Add assert.assertEquals validation

ITBOX-ITBOY commented:
Have to deal with

ITBOX-ITBOY commented:
> > > > @caishunfeng Please help to run the check program
> > > 
> > > 
> > > I've restarted the CI.
> > 
> > 
> > @EricGao888 There are some problems in the inspection. I have a look. I can't handle them
> 
> @ITBOX-ITBOY Did you try to run the UT locally for the http task?

@EricGao888 Please run the checker for me

EricGao888 commented:
> > > > > @caishunfeng Please help to run the check program
> 
> > > > 
> 
> > > > 
> 
> > > > I've restarted the CI.
> 
> > > 
> 
> > > 
> 
> > > @EricGao888 There are some problems in the inspection. I have a look. I can't handle them
> 
> > 
> 
> > @ITBOX-ITBOY Did you try to run the UT locally for the http task?
> 
> 
> 
> @EricGao888 Please run the checker for me

done

ITBOX-ITBOY commented:
> > > > > > @caishunfeng Please help to run the check program
> > 
> > 
> > > > >
> > 
> > 
> > > > >
> > 
> > 
> > > > > I've restarted the CI.
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > @EricGao888 There are some problems in the inspection. I have a look. I can't handle them
> > 
> > 
> > > 
> > 
> > 
> > > @ITBOX-ITBOY Did you try to run the UT locally for the http task?
> > 
> > 
> > @EricGao888 Please run the checker for me
> 
> done

@EricGao888 I don't know what problems to modify. I don't see any problems that need to be modified

ITBOX-ITBOY commented:
@SbloodyS Can you take a look? What do I need to deal with

caishunfeng commented on docs/docs/en/guide/parameter/context.md:
```diff
@@ -76,3 +76,21 @@ The result of Node_mysql is as follows:
+
+### HTTP
+
+Step 1: Drag an HTTP task, fill IN body for KEY, select OUT for IN/OUT, select VARCHAR for output data type, always select VARCHAR, nothing else.
```
If user can just select the `VARCHAR`, can it limit in UI?

ITBOX-ITBOY replied:
At present, we do not know how to deal with it. We will find a way to deal with it later

caishunfeng commented on docs/docs/zh/guide/parameter/context.md:
```diff
@@ -76,3 +77,22 @@ Node_mysql 运行结果如下：
+
+### HTTP
+
+拖一个http类型的任务，在自定义参数中KEY的位置填写body，在IN/OUT的部分选择OUT，在输出数据类型部分选择VARCHAR，一定要选择VARCHAR，不要选择别的哈。
```
Documentation should avoid colloquialism.
```suggestion
拖一个http类型的任务，在自定义参数中KEY的位置填写body，在IN/OUT的部分选择OUT，在输出数据类型部分选择VARCHAR（只能选择VARCHAR）。
```

ITBOX-ITBOY replied:
It has been processed and the code has been committed

caishunfeng commented on docs/docs/en/guide/parameter/context.md:
```diff
@@ -76,3 +76,21 @@ The result of Node_mysql is as follows:
+![httpParam-3](../../../../img/httpParam/httpParam-3.png)
+
+Step 3: You can write a test interface to test whether our parameters are passed successfully.
+![httpParam-4](../../../../img/httpParam/httpParam-4.png)
```
I think we don't need this in doc, WDYT? cc @zhongjiajie 

BTW, if add, it's better to use English log.

ITBOX-ITBOY replied:
It has been processed and the code has been committed

wangjiucheng123 committed:
Handle documentation issues

caishunfeng commented:
This pr should rebase the dev branch, avoid importing other commits.

wangjiucheng123 committed:
Merge branch 'apache:dev' into dev

wangjiucheng123 committed:
Resolve unread local variables

ITBOX-ITBOY commented on dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpParametersTest.java:
```diff
@@ -75,4 +81,36 @@
+        List<Property> varPool = new ArrayList<>();
+        httpParameters.setVarPool(varPool.toString());
+        httpParameters.dealOutParam(result);
+        Map<String, Property> varPoolMap = httpParameters.getVarPoolMap();
```
Has been resolved

wangjiucheng123 committed:
Merge branch 'apache:dev' into dev

wangjiucheng123 committed:
Merge branch 'dev' into dev-http-param

ITBOX-ITBOY commented:
@caishunfeng Please help to start the inspection procedure

wangjiucheng123 committed:
Merge branch 'apache:dev' into dev

Commit:
Merge branch 'dev' into dev-http-param

# Conflicts:
#	dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpParametersTest.java

Commit:
Resolve symbol not found problem

wangjiucheng123 committed:
Merge branch 'apache:dev' into dev

Commit:
Merge branch 'dev' into dev-http-param

Commit:
Handling check exceptions

wangjiucheng123 committed:
Merge branch 'apache:dev' into dev

Commit:
Merge branch 'dev' into dev-http-param

# Conflicts:
#	dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/main/java/org/apache/dolphinscheduler/plugin/task/http/HttpParameters.java
#	dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpParametersTest.java

wangjiucheng123 committed:
Merge branch 'apache:dev' into dev

Commit:
Merge branch 'dev' into dev-http-param

# Conflicts:
#	docs/docs/en/guide/parameter/context.md
#	docs/docs/zh/guide/parameter/context.md

wangjiucheng123 committed:
Merge branch 'apache:dev' into dev

Commit:
Merge branch 'dev' into dev-http-param

ITBOX-ITBOY commented:
@songjianet May I ask what I should modify to pass the inspection
